src/libraries/System.Private.CoreLib/src/System/Int16.cs

```diff
@@ -631,24 +631,11 @@ public static short Log2(short value)
         /// <inheritdoc cref="INumber{TSelf}.CopySign(TSelf, TSelf)" />
         public static short CopySign(short value, short sign)
         {
-            short absValue = value;
-
-            if (absValue < 0)
-            {
-                absValue = (short)(-absValue);
-            }
-
-            if (sign >= 0)
+            if (short.MinValue != value || 0 > sign)
             {
-                if (absValue < 0)
-                {
-                    Math.ThrowNegateTwosCompOverflow();
-                }
-
-                return absValue;
+                return unchecked((short)(value * Math.SignZeroToOne(value ^ sign)));
             }
-
-            return (short)(-absValue);
+            return checked((short)unchecked(-value)); // throws an OverflowException
```

Comment:
What about the exception message here?

Reply:
The custom message was saying that 'Negating the minimum value of a twos complement number is invalid.', which was added for this set of APIs(?), but it did not indicate the relation between the exception and the CopySign call. I think the default message is more understandable for the users. Using Math.ThrowNegateTwosCompOverflow() instead does not have a noticeable impact on the performance.